forch/faucetizer.py

```diff
@@ -141,11 +144,24 @@ def _augment_include_file_name(self, file_name):
         base_file_name, ext = os.path.splitext(file_name)
         return base_file_name + INCLUDE_FILE_SUFFIX + ext
 
-    def _is_access_port(self, port_cfg):
+    def _get_port_type(self, port_cfg):
+        testing_port_identifier = (self._config.orch_testing_config.testing_port_identifier or
+                                   TESTING_PORT_IDENTIFIER_DEFAULT)
+        if testing_port_identifier in port_cfg.get('description'):
+            return 'testing'
         non_access_port_properties = ['stack', 'lacp', 'output_only', 'tagged_vlans']
         port_properties = [
             property for property in non_access_port_properties if property in port_cfg]
-        return len(port_properties) == 0
+        return 'access' if len(port_properties) == 0 else 'other'
+
+    def _calcuate_available_tesing_vlans(self):
+        if not self._config.orch_testing_config.testing_vlans:
+            return
+        starting_vlan = self._config.orch_testing_config.testing_vlans[0]
```

Comment:
somewhere there needs to be a check that the stated vlan range does not overlap with any potential assigned segment vlans.

Reply:
Added the check in this function.

Reply:
Acutally moved the check into the initialization of Faucetizer...